modules/qualitycontrol/QualityController.ts

```diff
@@ -159,6 +161,26 @@ export class QualityController {
             (tpc: TraceablePeerConnection, stats: Map<number, IOutboundRtpStats>) => this._processOutboundRtpStats(tpc, stats));
     }
 
+    /**
+     * Creates a debounced function that delays the execution of the provided function until after the specified delay
+     * has elapsed. Unlike typical debounce implementations, the timer does not reset when the function is called again
+     * within the delay period.
+     *
+     * @param {Function} func - The function to be debounced.
+     * @param {number} delay - The delay in milliseconds.
+     * @returns {Function} - The debounced function.
+     */
+    _debounce(func: Function, delay: number) {
+        return function (...args) {
+            if (!this._timer) {
+                this._timer = setTimeout(() => {
+                    func.apply(this, args);
+                    this._timer = null;
```

Comment:
If we end up keeing this, I'd reset the timer first, or it will get stuck if the function throws an exception.